### ISSUE TITLE
Rescheduler not running?

### ISSUE THREAD

**Issue by dcowden** [kubernetes/kops#4110]
[According to documentation](https://v1-7.docs.kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) since k8s version 1.7, by default a static rescheduler pod should be running, but in our cluster it is not. The rescheduler should be making sure that critical pods are running. 

Kops appears to enable it by default [here](https://github.com/kubernetes/kops/blob/master/vendor/k8s.io/kubernetes/cluster/gce/config-default.sh#L220), and we've not changed anything that should disable it. 

Does anyone know what's necessary to get it running? We can run it manually using [this manifest](https://github.com/kubernetes/kubernetes/blob/master/cluster/saltbase/salt/rescheduler/rescheduler.manifest), but we don't want to do that unless we know for sure kops isnt running it already for some good reason.

**Operating Details:**
kops 1.8.0
k8s version 1.7.9, on AWS, with CentOS images

**Comment by chrislovecnm** [comment]
@dcowden interesting ... @justinsb any reason we are not running this??

**Comment by dcowden** [comment]
For now, we ran it manually using the manifest above. We needed to adjust the volume mounts for pre 1.8 syntax, but other than that it is running ok. We're in the middle of doing some testing, and if we find out something interesting we'll report back.

We started with a 3 node cluster, and a 2-replicas kube-dns installation working great. Over the course of two or three rolling updates, we ended up having workloads affected due to inability to resolve names. We eventually found out that over time, the two kube-dns pods had worked their way onto a single node, and finally that node was upgraded.  

Not the subject of this issue, but worth noting, is that it appears that kops doesn't set up PodDisruptionBudgets for kube-dns either-- so draining the node doesnt seem to force k8s to reschedule those before returning.

We're hoping that a combination of setting up proper PodDisruptionBudgets and the rescheduler will prevent this from happening.
